pandas/core/arrays/_mixins.py

```diff
@@ -128,17 +128,31 @@ def view(self, dtype: Dtype | None = None) -> ArrayLike:
         dtype = pandas_dtype(dtype)
         arr = self._ndarray
 
-        if isinstance(dtype, (PeriodDtype, DatetimeTZDtype)):
+        if isinstance(dtype, PeriodDtype):
             cls = dtype.construct_array_type()
             return cls(arr.view("i8"), dtype=dtype)
+        elif isinstance(dtype, DatetimeTZDtype):
+            # error: Incompatible types in assignment (expression has type
+            # "type[DatetimeArray]", variable has type "type[PeriodArray]")
+            cls = dtype.construct_array_type()  # type: ignore[assignment]
+            dt64_values = arr.view(f"M8[{dtype.unit}]")
+            return cls(dt64_values, dtype=dtype)
         elif dtype == "M8[ns]":
```

Comment:
Why do just the `ns` cases need special casing?

Reply:
i also have a branch to extend this check to other supported dt64/td64 dtypes, just trying to keep this branch targeted